### PR TITLE
fix: Path issues with pip modules

### PR DESCRIPTION
**Pull request by tokotchd**
Always prepend current bentofile directory to system path to avoid unwanted behavior when other bentofiles are on the system PATH
This is especially evident when trying to use bentoml cli on python folders that are also pip modules see https://github.com/bentoml/BentoML/issues/4217

### CODE REVIEW

**Review by sauyon — APPROVED** [review]
LGTM, thank you!

File: src/bentoml/_internal/service/loader.py [hunk 1]
```diff
@@ -172,7 +169,7 @@ def recover_standalone_env_change():
         object.__setattr__(instance, "_import_str", f"{module_name}:{attrs_str}")
         return instance
     except ImportServiceError:
-        if sys_path_modified and working_dir:
+        if working_dir:
```
Comment (parano) [inline src/bentoml/_internal/service/loader.py:172]: we probably don't need this check neither now?

Comment (sauyon) [reply src/bentoml/_internal/service/loader.py:172]: I think this check is ensuring that if the assignment fails it doesn't try to remove `None`.

Comment (parano) [reply src/bentoml/_internal/service/loader.py:172]: Make sense, I'd suggest we move the working_dir assignment L75-81 out of the `try` block and remove this `if working_dir` check as well

Comment (parano) [reply src/bentoml/_internal/service/loader.py]: @sauyon made a quick patch, could you help take a look?

**Pull request closed without merging**